test/lsp.py

```diff
@@ -1296,6 +1308,40 @@ def user_interaction_failed_autoupdate(self, test, sub_dir):
 
     # {{{ actual tests
 
+    def test_analyze_all_project_files(self, solc: JsonRpcProcess) -> None:
+        """
+        Tests the option (default) to analyze all .sol project files even when they have not been actively
+        opened yet. This is how other LSP's (at least for C++) work too and it makes cross-unit tasks
+        actually correct (e.g. symbolic rename, find all references, ...).
+
+        In this test, we simply open up a custom project and ensure we're receiving the diagnostics
+        for all existing files in that project (while having none of these files opened).
+        """
+        SUBDIR = 'analyze-full-project'
+        self.setup_lsp(
+            solc,
+            analyze_all_files_in_project=True,
+            project_root_suffix=SUBDIR
+        )
+        published_diagnostics = self.wait_for_diagnostics(solc)
+        pprint(published_diagnostics)
+        self.expect_equal(len(published_diagnostics), 3, "Diagnostic reports for 3 files")
+
+        # C.sol
+        report = published_diagnostics[0]
+        self.expect_equal(report['uri'], self.get_test_file_uri('C', SUBDIR), "Correct file URI")
+        self.expect_equal(len(report['diagnostics']), 0, "no diagnostics")
+
+        # D.sol
+        report = published_diagnostics[1]
+        self.expect_equal(report['uri'], self.get_test_file_uri('D', SUBDIR), "Correct file URI")
+        self.expect_equal(len(report['diagnostics']), 0, "no diagnostics")
+
+        # E.sol
+        report = published_diagnostics[2]
```

Comment:
You should also check that no other files were found.

Reply:
That's done like 12 lines above. :)

Reply:
Sorry, my bad. It was actually there.